packages/core/test/unit/segments/attributes/remote_request_data.test.js

```diff
@@ -50,5 +50,14 @@ describe('RemoteRequestData', function() {
         ''
       );
     });
+    it('should use the host from the request object over headers', () => {
+      const requestWithHost = Object.assign(request, { host: 'different-site.com' });
```

Comment:
Hmm - could cleanup in another PR, `getHeader() { return 'host.com' }` seems like very suspicious code :) But I guess that's what makes sure we're overwriting the header.

By the way, if not in a unit test is it possible to verify the behavior with `fetch` which was the example in the filed issue? I assume it's `node-fetch`. Don't see a reason it wouldn't work though

https://github.com/node-fetch/node-fetch/blob/master/src/request.js#L211

Reply:
I updated the `getHeader` to be less sus :) 

As for testing `node-fetch`, we could and probably should, but that would require a whole lot more stubbing than is probably in scope for this change since we're only testing the underlying Node HTTP client today.